model-optimizer/extensions/middle/UnsqueezeTileReshapeBlockToInterpolate.py

```diff
@@ -76,79 +105,107 @@ def pattern(self):
-        scales_node.out_port(0).connect(mul_node.in_port(1))
-
-        slice_begin = Const(graph, dict(name=unsqueeze_name + '/slice_begin', value=int64_array([axis]))).create_node()
-        slice_end = Const(graph, dict(name=unsqueeze_name + '/slice_end', value=int64_array([axis + 1]))).create_node()
-
-        strided_slice_node = StridedSlice(graph,
-                                          {'name': unsqueeze_name + '/StridedSlice',
-                                           'begin_mask': int64_array([1]),
-                                           'end_mask': int64_array([1]),
-                                           'new_axis_mask': int64_array([0]),
-                                           'shrink_axis_mask': int64_array([0]),
-                                           'ellipsis_mask': int64_array([0]),
-                                           }).create_node()
-        shape_node.out_port(0).connect(strided_slice_node.in_port(0))
-        slice_begin.out_port(0).connect(strided_slice_node.in_port(1))
-        slice_end.out_port(0).connect(strided_slice_node.in_port(2))
+        expected_tile_constant = np.ones(input_rank_of_unsqueeze + 1)
+        expected_tile_constant[d_idx] = float(second_input_of_tile.value[d_idx])
 
-        cast_shape_to_float = Cast(graph, {'dst_type': np.float32}).create_node()
+        if not np.array_equal(expected_tile_constant, float32_array(second_input_of_tile.value)):
+            return False
+
+        reshape_node = match['reshape']
+        new_shape = reshape_node.in_port(1).data.get_value()
+        if new_shape is None or input_rank_of_unsqueeze != len(new_shape):
+            return False
 
-        strided_slice_node.out_port(0).connect(cast_shape_to_float.in_port(0))
-        cast_shape_to_float.out_port(0).connect(mul_node.in_port(0))
+        return True
 
-        interp_node = Interpolate(graph,
-                                  dict(mode='nearest',
-                                       antialias=0, pads_begin=int64_array([0]),
-                                       pads_end=int64_array([0]), coordinate_transformation_mode='half_pixel',
-                                       nearest_mode='round_prefer_floor', cube_coeff=-0.75,
-                                       version='opset4', shape_calculation_mode='scales',
-                                       in_ports_count=4,
-                                       maybe_part_of_sequence=True)).create_node()
+    def replace_pattern(self, graph: Graph, match: dict):
+        if not self.is_applicable(match):
+            return
 
-        floor_node = Floor(graph, {'name': unsqueeze_name + '/Floor'}).create_node()
-        cast_mul_result_to_int = Cast(graph, {'dst_type': np.int64}).create_node()
+        unsqueeze_node = match['unsqueeze']
+        unsqueeze_name = unsqueeze_node.soft_get('name', unsqueeze_node.id)
+        second_input_of_unsqueeze = unsqueeze_node.in_port(1).get_connection().get_source().node
```

Comment:
I think we need to check that the second input of Tile is a constant. Otherwise, the model can get no longer reshape-able.

Reply:
There is a check for this already:

`if not second_input_of_tile.has_valid('value'):`

which will be true only if the `second_input_of_tile` is Const

Reply:
This is a middle phase transformation and it can have a value if the node is under `ShapeOf`. Please correct me if I am wrong.

Reply:
`second_input_of_tile` is an operation node (not data node). Operation node with attribute 'value' are only Const nodes.

Reply:
Ok, I buy it but I would rather check `op` attribute explicitly.